### PR TITLE
Add handlers for gsm, network, power and sms command

### PR DESCRIPTION
**Pull request by SrinivasanTarget**
## Change list

Add handlers for gsm, network, power and sms command
https://github.com/appium/appium-base-driver/blob/master/lib/protocol/routes.js#L366-L375
 
## Types of changes

What types of changes are you proposing/introducing to Java client?
_Put an `x` in the boxes that apply_

- [ ] No changes in production code.
- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

@mykola-mokhnach ping

### CODE REVIEW

File: src/main/java/io/appium/java_client/android/GsmCallActions.java [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+
+public enum GsmCallActions {
+    CALL("call"),
+    ACCEPT("accept"),
+    CANCEL("cancel"),
+    HOLD("hold");
+
+    private final String gsmcall;
+
+    GsmCallActions(String gsmcall) {
```
Comment (mykola-mokhnach) [inline src/main/java/io/appium/java_client/android/GsmCallActions.java]: this constructor is redundant. All enum elements have `name()` and `ordinal()` properties. That's pretty all you need. Same comment to other enums

File: src/main/java/io/appium/java_client/android/AndroidDriver.java [hunk 2]
```diff
@@ -176,4 +183,68 @@ public void toggleLocationServices() {
         CommandExecutionHelper.execute(this, toggleLocationServicesCommand());
     }
 
+    /**
```
Comment (mykola-mokhnach) [inline src/main/java/io/appium/java_client/android/AndroidDriver.java]: I'd rather put these methods into separate interface named as `SupportsChangingEmulatorState`

File: src/main/java/io/appium/java_client/android/AndroidMobileCommandHelper.java [hunk 3]
```diff
@@ -291,4 +291,109 @@
...
+
+    /**
+     * This method forms a {@link Map} of parameters for the element
+     * value replacement. It is used against input elements
+     *
+     * @param powerACState One of available power AC state
+     *
+     * @return a key-value pair. The key is the command name. The value is a {@link Map} command arguments.
+     */
+    public static Map.Entry<String, Map<String, ?>>  powerACCommand(
```
Comment (mykola-mokhnach) [inline src/main/java/io/appium/java_client/android/AndroidMobileCommandHelper.java]: all these methods ahev one extra space between the return type and the method name

File: src/main/java/io/appium/java_client/android/AndroidDriver.java [hunk 4]
```diff
@@ -176,4 +183,68 @@ public void toggleLocationServices() {
...
+     * @param percent   Percentage value in range [0, 100].
+     */
+    public void powerCapacity(int percent) {
+        CommandExecutionHelper.execute(this, powerCapacityCommand(percent));
+    }
+
+    /**
+     * Emulate power state change on the connected emulator.
+     *
+     * @param powerACState   One of available Power AC state.
```
Comment (mykola-mokhnach) [inline src/main/java/io/appium/java_client/android/AndroidDriver.java]: I'd rather put a direct link to the corresponding enum here

**Comment by SrinivasanTarget** [comment]
> It looks like we need to add at least one assert to each test to make Codacy happy

Yeah but i'm not sure what to assert here? Do you have any better suggestion?

**Comment by mykola-mokhnach** [comment]
```
try {
  smth();
} catch (Exception e) {
  fail(e)
}
```

try maybe this workaround

File: src/test/java/io/appium/java_client/android/AndroidDriverTest.java [hunk 5]
```diff
@@ -38,29 +38,53 @@
 public class AndroidDriverTest extends BaseAndroidTest {
 
     @Test public void sendSMSTest() {
-        driver.sendSMS("11111111", "call");
+        try {
+            driver.sendSMS("11111111", "call");
+        } catch (Exception e) {
+           assertFalse( "method works only in emulators", true);
```
Comment (mykola-mokhnach) [inline src/test/java/io/appium/java_client/android/AndroidDriverTest.java]: You can use `fail` call instead

File: src/main/java/io/appium/java_client/android/SupportsSpecialEmulatorCommands.java [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+        CommandExecutionHelper.execute(this, sendSMSCommand(phoneNumber, message));
+    }
+
+    /**
+     * Emulate GSM call event on the connected emulator.
+     *
+     * @param phoneNumber The phone number of the caller.
+     * @param gsmCallActions   One of available {@link GsmCallActions} values.
+     */
+    default void setGsmCall(String phoneNumber, GsmCallActions gsmCallActions) {
```
Comment (mykola-mokhnach) [inline src/main/java/io/appium/java_client/android/SupportsSpecialEmulatorCommands.java]: `makeGsmCall` would be a better name

File: src/main/java/io/appium/java_client/android/SupportsSpecialEmulatorCommands.java [hunk 7]
```diff
@@ -0,0 +1,81 @@
...
+        CommandExecutionHelper.execute(this, gsmCallCommand(phoneNumber, gsmCallActions));
+    }
+
+    /**
+     * Emulate GSM signal strength change event on the connected emulator.
+     *
+     * @param gsmSignalStrength   One of available {@link GsmSignalStrength} values.
+     */
+    default void setGsmSignalStrength(GsmSignalStrength gsmSignalStrength) {
+        CommandExecutionHelper.execute( this, gsmSignalStrengthCommand(gsmSignalStrength));
```
Comment (mykola-mokhnach) [inline src/main/java/io/appium/java_client/android/SupportsSpecialEmulatorCommands.java]: extra space before `this`

**Pull request merged**